src/WebJobs.Script.WebHost/Configuration/HostJsonFileConfigurationSource.cs

```diff
@@ -0,0 +1,243 @@
+                // for all other logging after startup.
+                // TODO: DI (FACAVAL) Fix this
+                //ConfigureLoggerFactory();
+
+                // TODO: DI (FACAVAL) Logger configuration to move to startup:
+                // Logger = _startupLogger = _hostOptions.LoggerFactory.CreateLogger(LogCategories.Startup);
+                ScriptWebHostOptions options = _configurationSource.HostOptions.Value;
+                string hostFilePath = Path.Combine(options.ScriptPath, ScriptConstants.HostMetadataFileName);
+                string readingFileMessage = string.Format(CultureInfo.InvariantCulture, "Reading host configuration file '{0}'", hostFilePath);
+                JObject hostConfigObject = LoadHostConfig(hostFilePath);
+                string sanitizedJson = SanitizeHostJson(hostConfigObject);
+                string readFileMessage = $"Host configuration file read:{NewLine}{sanitizedJson}";
+
+                // TODO: DI (FACAVAL) See method comments in ScriptHost
+                //ApplyConfiguration(hostConfigObject, ScriptConfig, _startupLogger);
+
+                // TODO: DI (FACAVAL) Review - this should likely move and not be just config
+                //if (_settingsManager.FileSystemIsReadOnly)
+                //{
+                //    // we're in read-only mode so source files can't change
+                //    ScriptConfig.FileWatchingEnabled = false;
+                //}
+
+                // now the configuration has been read and applied re-create the logger
+                // factory and loggers ensuring that filters and settings have been applied
+                // TODO: DI (FACAVAL) TODO
+                //ConfigureLoggerFactory(recreate: true);
+
+                // TODO: DI (FACAVAL) Logger configuration to move to startup
+                //_startupLogger = _hostOptions.LoggerFactory.CreateLogger(LogCategories.Startup);
+                //Logger = _hostOptions.LoggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+
+                // Allow tests to modify anything initialized by host.json
+                //ScriptConfig.OnConfigurationApplied?.Invoke(ScriptConfig);
+                _logger.LogTrace("Host configuration applied.");
+
+                // Do not log these until after all the configuration is done so the proper filters are applied.
+                _logger.LogInformation(readingFileMessage);
+                _logger.LogInformation(readFileMessage);
+
+                // TODO: DI (FACAVAL) Move this to a more appropriate place
+                // If they set the host id in the JSON, emit a warning that this could cause issues and they shouldn't do it.
+                if (hostConfigObject["id"] != null)
+                {
+                    _logger.LogWarning("Host id explicitly set in the host.json. It is recommended that you remove the \"id\" property in your host.json.");
+                }
+
+                // TODO: DI (FACAVAL) Move to options setup
```

Comment:
There is a bunch of commented out code in here with TODOs - what's the plan for resolving all these?

Reply:
I'm actively working on remaining configuration items (both in core and here), so this is getting cleaned up. A lot of what is commented out at the moment is for validation when that work is done.